cryptographic_flows.md

```diff
@@ -1,27 +1,24 @@
 
-# Authenticate
-The local client inputs a user ID and a password and receives an "export key", which we will consider as a valid symmetric encryption key, as output. The key server receives as output some user-specific authentication material, which it can compare to its stored data to verify authentication.
-
-Both parties recieve an open channel as output, which satisfies mutual authentication, confidentiality, and integrity.
-
-
 # Generate a secret
 
-The local client component can generate a secret. Using a cryptographically secure pseudo-random number generator with a random seed, generate 256 bits of output.
-
-Generate a unique tag to associate with the secret (unique among secrets stored by the local client).
+The local client component can generate a secret.
+1. Generate the secret: using a cryptographically secure pseudo-random number generator with a random seed, generate 256 bits of output.
+2. Generate a unique tag to associate with the secret (unique among secrets stored by the local client). A tag is 16 bytes (128 bits) of arbitrary data. It can be generated deterministically (e.g. incrementing by 1 for each new secret) or pseudorandomly.
+3. Output the tag and the secret.
 
 # Generate and store a secret locally
-The local client component can run the [generate](generate-a-secret) protocol. Then, store the tag and the secret as a pair.
+The local client component can run this operation.
+1. Execute the [generate](#generate-a-secret) protocol.
+2. Store the tag and the secret as a pair.
```

Comment:
L12 is too vague: requirements for local storage for the client should be captured as an issue and linked in a TODO comment.